core/sodasql/telemetry/soda_telemetry.py

```diff
@@ -0,0 +1,93 @@
+import hashlib
+import json
+import logging
+import platform
+from typing import Dict
+
+
+from opentelemetry import trace
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.semconv.resource import ResourceAttributes
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import (
+    BatchSpanProcessor,
+    ConsoleSpanExporter
+)
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from requests.sessions import session
+
+from sodasql.__version__ import SODA_SQL_VERSION
+from sodasql.common.config_helper import ConfigHelper
+from sodasql.scan.dialect import Dialect
+
+logger = logging.getLogger(__name__)
+soda_config = ConfigHelper.get_instance()
+
+class SodaTelemetry:
+    """Main entry point for Open Telemetry tracing.
+
+    For more info about what and why visit https://github.com/sodadata/soda-sql/issues/543.
+
+    The main goal of this class is to concentrate as much tracing data and logic as reasonable.
+    This code design means that compromises have been made on code design in order to facilitate maximum
+    transparency and avoid scattering tracing code around the codebase.
+
+    With that being said, some tracing is still present in other files, e.g.:
+    - `/core/sodasql/cli/cli.py` - CLI arguments and options tracing
+    - `/core/sodasql/scan/warehouse.py` - safe datasource type and hash tracing
+
+    This list is not necessarily exhaustive, search for `from sodasql.telemetry.soda_telemetry import soda_telemetry` imports OR
+    `set_attribute` method usage to obtain the full list.
+    """
+
+    ENDPOINT = 'https://collect.dev.sodadata.io/v1/traces'
+    __skip = False
+    soda_config = ConfigHelper.get_instance()
+
+    def __init__(self,  skip: bool = False):
+        self.__skip = skip
+        if self.__skip:
+            logger.info("Skipping usage telemetry.")
+        else:
+            logger.info("Sending usage telemetry.")
+            self.__setup()
+
+    def __setup(self):
+        session
+
+        provider = TracerProvider(
+            resource=Resource.create(
+                {
+                    'os.architecture': platform.architecture(),
+                    'python.version': platform.python_version(),
+                    'python.implementation': platform.python_implementation(),
```

Comment:
would it be possible to use _s here when I flatten this into a datawarehouse table it'll be painful probably. Unless I'm missing some reasoning?

Reply:
I meant whether it would be possible to use underscores instead of dots in the key names so that it is easier when we flatten the json to table columns. I'm not entirely sure how well snowflake copes with dots as column names.

Reply:
Thanks for checking on snowflake @m1n0 

Regarding the second idea. At this point, I'm not too sure, as I have not looked enough into how I will flatten. My idea would be I'd flatten as simple columns yes, but I prefer the idea of the structured approach from a "type" perspective, however, I don't know if that makes tings easy or not.

At this point let's do the following:
- leave the dots
- I need to play around with it which I should be able to do within a day or so of setting up a pipe.
- if it's annoying, we can choose to break the best practice or I place some aliasing in Transform stage or maybe the structured approach works.

Since I don't have a good answer, let's add to the current scope and go with that. If we need to make a change it will be small I imagine.